src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. You should restrict the type parameter `T` to extend `Machine` or its subclasses.